tests/core/failover/task.yaml

```
@@ -61,21 +89,41 @@ execute: |
         truncate -s 0 "$BUILD_DIR/unpack/initrd.img"
     }
 
+    if os.query is-core18 && [[ "$SPREAD_VARIANT" = "rclocalcrash" ]]; then
+        # there is no /etc/rc.local on core18
+        echo "scenario isn't supported on core18"
+        exit 0
+    fi
+
     #shellcheck source=tests/lib/names.sh
     . "$TESTSLIB"/names.sh
     #shellcheck source=tests/lib/snaps.sh
     . "$TESTSLIB"/snaps.sh
+    #shellcheck source=tests/lib/store.sh
+    . "$TESTSLIB"/store.sh
+
+    core_name="core"
+    core_snap_id="$SNAP_ID_core"
+    if os.query is-core18; then
+        core_name="core18"
+        core_snap_id="$SNAP_ID_core18"
+    elif os.query is-core20; then
+        core_name="core20"
+        core_snap_id="$SNAP_ID_core20"
+    fi
 
     if [ "$TARGET_SNAP" = kernel ]; then
-        TARGET_SNAP_NAME=$kernel_name
+        TARGET_SNAP_NAME=pc-kernel
+        SNAP_ID="$SNAP_ID_pc_kernel"
     else
-        TARGET_SNAP_NAME=core
+        TARGET_SNAP_NAME="$core_name"
+        SNAP_ID="$core_snap_id"
     fi
 
     if [ "$SPREAD_REBOOT" = 0 ]; then
         # first pass, save current target snap revision
         snap list | awk "/^${TARGET_SNAP_NAME} / {print(\$3)}" > prevBoot
-
+        START_REVISION=$(($(cat prevBoot) + 1))
```

I see the test failing with this error in that line, it is needed to calculate the rev from the store.
/bin/bash: line 142: x1: unbound variable

---

It's a bit complex, but this will work:
```suggestion
        PREV_REVISION=$(cat prevBoot)
        START_REVISION=$(printf "%s%d" ${PREV_REVISION%[0-9]} $((${PREV_REVISION##[^0-9]}+1)))
```
Or we can split it to make it a bit more readable:
```suggestion
        PREV_REVISION=$(cat prevBoot)
        PREV_REVISION_TEXT=${PREV_REVISION%[0-9]}
        PREV_REVISION_NUMBER=${PREV_REVISION##[^0-9]}
        START_REVISION=$(printf "%s%d" ${PREV_REVISION_TEXT} $((${PREV_REVISION_NUMBER}+1)))
```

---

x1 is a local revision it's unexpected that it appears here.